### PR TITLE
[DOCS] Rewrite `fuzzy` query docs

### PR DESCRIPTION
**Pull request by jrodewig**
## Changes
* Rewrites description of `fuzzy` query
* Include all parameters in example request
* Adds parameters sections

This is part of #40977, an effort to standardize documentation for query types.

Any and all feedback welcome!

## Before
<details>
 <summary>Before image</summary>
Screen Shot 2019-05-08 at 11 34 54 AM
</details>

## After
<details>
 <summary>After image</summary>
Screen Shot 2019-05-08 at 11 36 13 AM
</details>

### CODE REVIEW

**Comment by JessePelton** [comment]
"Removing a character: shouldn't "**b**lack -> back" be either "b**l**ack -> back" or "**b**lack -> lack"? (The latter is easier to discern visually.)

**Comment by jrodewig** [comment]
Thanks for the excellent catch @JessePelton! I've updated the example with 222eccd.

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/docs-check

File: docs/reference/query-dsl/fuzzy-query.asciidoc [hunk 1]
```diff
@@ -1,75 +1,99 @@
 [[query-dsl-fuzzy-query]]
 === Fuzzy Query
 
-The fuzzy query uses similarity based on Levenshtein edit distance.
+Returns documents that contain terms similar to the search term.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/fuzzy-query.asciidoc]: as "similar" may mean different things even within ES, may be better to say "contain terms similar to the search term as measured by Levenshtein edit distance"

File: docs/reference/query-dsl/fuzzy-query.asciidoc [hunk 2]
```diff
@@ -1,75 +1,99 @@
...
 
-==== String fields
+{es} measures similarity, or fuzziness, using a
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/fuzzy-query.asciidoc]: {es} measures similarity -> may be skip this sentence, or somehow indicate that  for this particular  query "similarity" (not a general notion of similarity in es).

File: docs/reference/query-dsl/fuzzy-query.asciidoc [hunk 3]
```diff
@@ -1,75 +1,99 @@
...
-    "query": {
-       "fuzzy" : { "user" : "ki" }
-    }
-}
---------------------------------------------------
-// CONSOLE
-
-Or with more advanced settings:
+[[fuzzy-query-ex-request]]
+==== Example request
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/fuzzy-query.asciidoc]: Kind of nice to have a short version example as well, as I imagine that what most users are using

File: docs/reference/query-dsl/fuzzy-query.asciidoc [hunk 4]
```diff
@@ -1,75 +1,99 @@
...
+`2`:: Terms must match within two edits.
+--
 
 `max_expansions`::
++
+--
+(Optional, integer) Maximum number of variations created. Defaults to `50`.
++
+WARNING: Avoid using a high value in the `max_expansions` parameter, especially
+if the `transpositions` parameter value is `0`. High values in the
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/fuzzy-query.asciidoc]: if the `transpositions` parameter value is `0`?
 Did you mean if `prefix_length` is set to `0`?

File: docs/reference/query-dsl/fuzzy-query.asciidoc [hunk 5]
```diff
@@ -1,75 +1,99 @@
...
+(Required, string) Term you wish to find in the provided `<field>`.
 
-    The maximum edit distance. Defaults to `AUTO`. See <<fuzziness>>.
-
-`prefix_length`::
-
-    The number of initial characters which will not be ``fuzzified''. This
-    helps to reduce the number of terms which must be examined. Defaults
-    to `0`.
+`fuzziness`::
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/fuzzy-query.asciidoc]: fuzziness is used in more than one query. That's why it is good to keep its definition in a single separate documents that other queries can refer to.  It is also would be easier to change a single document if something is changed. Is there a reason to expand this definition here?

**Review by mayya-sharipova — APPROVED** [review]
@jrodewig Thanks for the PR. I have left some comments.

File: docs/reference/query-dsl/fuzzy-query.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/query-dsl/fuzzy-query.asciidoc]: Fixed with 9b97c9e3c00.

File: docs/reference/query-dsl/fuzzy-query.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/query-dsl/fuzzy-query.asciidoc]: Removed with 9b97c9e.

File: docs/reference/query-dsl/fuzzy-query.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/query-dsl/fuzzy-query.asciidoc]: Re-added a simple example with 7b5e10aa887.

File: docs/reference/query-dsl/fuzzy-query.asciidoc (hunk 4 above)
Comment (jrodewig) [reply docs/reference/query-dsl/fuzzy-query.asciidoc]: Thank you for catching this! Corrected with 7a1daa30fb1.

File: docs/reference/query-dsl/fuzzy-query.asciidoc (hunk 5 above)
Comment (jrodewig) [reply docs/reference/query-dsl/fuzzy-query.asciidoc]: This is a great point. I replaced this with a cross-reference in 9cec50a093e.

**Comment by jrodewig** [comment]
@elasticmachine  update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Pull request merged**